examples/Spec/Int.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Spec.Int (HasLogicalModel (..), IntProp(..),intGenTests,intPureTests,intPlutarchTests) where
+import Proper.HasLogicalModel
+import Proper.LogicalModel
+import Proper.HasParameterisedGenerator
+import Proper.HasPlutusTestRunner
+import Proper.HasPureTestRunner
+import SAT.MiniSat ( Formula (..) )
+import Hedgehog (forAll)
+import qualified Hedgehog.Gen as Gen
+import Hedgehog.Range (linear)
+import Data.Proxy (Proxy(..))
+import Test.Tasty (TestTree,testGroup)
+import Test.Tasty.Hedgehog (fromGroup)
+
+import Plutarch (compile)
+import Plutarch.Prelude
+
+data IntProp =
+      IsNegative
+    | IsPositive
+    | IsZero
+    | IsLarge
+    | IsSmall
+    | IsMaxBound
+    | IsMinBound
+    deriving stock (Eq,Ord,Enum,Show,Bounded)
+
+instance LogicalModel IntProp where
+  logic = ExactlyOne [Var IsNegative, Var IsPositive, Var IsZero]
+     :&&: ExactlyOne [Var IsLarge, Var IsSmall]
+     :&&: (Var IsZero :->: Var IsSmall)
+     :&&: (Var IsMaxBound :->: (Var IsLarge :&&: Var IsPositive))
+     :&&: (Var IsMinBound :->: (Var IsLarge :&&: Var IsNegative))
```

Comment:
We can start simply with a set of Propositions and associated model logic. This logic defines which Propositions are valid in conjunction. For example `IsNegative` and `IsPositive` cannot both be true simultaneously.